maestro-ios/src/main/java/ios/CompositeIOSDevice.kt

```diff
@@ -0,0 +1,127 @@
+import ios.xctest.XCTestIOSDevice
+import okio.Sink
+import java.io.File
+import java.io.InputStream
+
+class CompositeIOSDevice(
+    override val deviceId: String?,
+    private val idbIOSDevice: IOSDevice,
+    private val xcTestDevice: IOSDevice,
+) : IOSDevice {
+
+    override fun open() {
+        idbIOSDevice.open()
+        xcTestDevice.open()
+    }
+
+    override fun deviceInfo(): Result<DeviceInfo, Throwable> {
+        return idbIOSDevice.deviceInfo()
+    }
+
+    override fun contentDescriptor(): Result<XCUIElement, Throwable> {
+        return xcTestDevice.contentDescriptor()
+            .recoverIf(
+                { it is XCTestIOSDevice.IllegalArgumentSnapshotFailure },
+                {
+                    idbIOSDevice.contentDescriptor()
+                        .getOrThrow()
+                }
+            )
+    }
+
+    override fun tap(x: Int, y: Int): Result<Unit, Throwable> {
+        return idbIOSDevice.tap(x, y)
+    }
+
+    override fun longPress(x: Int, y: Int): Result<Unit, Throwable> {
+        return idbIOSDevice.longPress(x, y)
+    }
+
+    override fun pressKey(code: Int): Result<Unit, Throwable> {
+        return idbIOSDevice.pressKey(code)
+    }
+
+    override fun pressButton(code: Int): Result<Unit, Throwable> {
+        return idbIOSDevice.pressButton(code)
+    }
+
+    override fun scroll(
+        xStart: Float,
+        yStart: Float,
+        xEnd: Float,
+        yEnd: Float,
+        velocity: Float?
+    ): Result<Unit, Throwable> {
+        return xcTestDevice.scroll(xStart, yStart, xEnd, yEnd, velocity)
+    }
+
+    override fun input(text: String): Result<Unit, Throwable> {
+        return xcTestDevice.input(text)
+    }
+
+    override fun install(stream: InputStream): Result<Unit, Throwable> {
+        return idbIOSDevice.install(stream)
+    }
+
+    override fun uninstall(id: String): Result<Unit, Throwable> {
+        return idbIOSDevice.uninstall(id)
+    }
+
+    override fun pullAppState(id: String, file: File): Result<Idb.PullResponse, Throwable> {
+        return idbIOSDevice.pullAppState(id, file)
+    }
+
+    override fun pushAppState(id: String, file: File): Result<Unit, Throwable> {
+        return idbIOSDevice.pushAppState(id, file)
+    }
+
+    override fun clearAppState(id: String): Result<Idb.RmResponse, Throwable> {
+        return idbIOSDevice.clearAppState(id)
+    }
+
+    override fun clearKeychain(): Result<Unit, Throwable> {
+        return idbIOSDevice.clearKeychain()
+    }
+
+    override fun launch(id: String): Result<Unit, Throwable> {
+        return idbIOSDevice.launch(id)
+    }
+
+    override fun stop(id: String): Result<Unit, Throwable> {
+        return idbIOSDevice.stop(id)
+    }
+
+    override fun openLink(link: String): Result<Unit, Throwable> {
+        return idbIOSDevice.openLink(link)
+    }
+
+    override fun takeScreenshot(out: Sink): Result<Unit, Throwable> {
+        return idbIOSDevice.takeScreenshot(out)
```

Comment:
This was not happening with idb 🤔 implementation communicates with `XCRunnerSimctl` which directly issues simctl commands. Can we remove this idbIOSDevice call completely and directly use the XCRunnerSimctl.

Reply:
Whoops, will fix that by introducing a `XCRunDevice` and calling that one